js/ui/paygUnlockDialog.js

```diff
@@ -328,6 +333,53 @@ var PaygUnlockDialog = new Lang.Class({
         }
     },
 
+    _onSuccess: function(mainBox) {
+            mainBox.remove_all_children();
+
+            //Measuring the ceiling of the timeRemainingSecs because by the time this runs, some time has already elapsed
+            let timeString = Main.paygManager._convertTimeToString(Math.ceil(Main.paygManager._timeRemainingSecs()));
+            let successLabel = new St.Label({ style_class: 'unlock-dialog-payg-title',
+                                        text: _(timeString + " have been added to your PayGo credit."),
+                                        x_align: Clutter.ActorAlign.CENTER });
+            mainBox.add_child(successLabel);
+
+            let successButtonBox = new St.BoxLayout({ style_class: 'unlock-dialog-payg-button-box',
+                                            vertical: false,
+                                            x_expand: true,
+                                            x_align: Clutter.ActorAlign.FILL,
+                                            y_expand: true,
+                                            y_align: Clutter.ActorAlign.END });
+
+            this._successButton = new St.Button({ style_class: 'modal-dialog-button button',
+                                           button_mask: St.ButtonMask.ONE | St.ButtonMask.THREE,
+                                           reactive: true,
+                                           can_focus: true,
+                                           label: _("Success!"),
+                                           x_align: St.Align.CENTER,
+                                           y_align: St.Align.CENTER });
+
+            this._successButton.add_style_pseudo_class('default');
+            successButtonBox.add_child(this._successButton);
+
+            mainBox.add_child(successButtonBox);
+
+            //Creating a variable to control the trigger to the signal whether or not it`s
+            //coming from a button, since they are asynchronous.
+            let successButtonClicked = false;
+            log ('Trigger deactivation of ScreenShield');
+            this._successButton.connect('clicked', () => {
+                successButtonClicked = true;
+                Main.paygManager.emit('success-on-expiry-changed');
```

Comment:
I don't like the name of this signal. I'd prefer `credit-added` or something like that. And it seems very dodgy that the unlock dialog is emitting signals on the manager.